src/App.tsx

```diff
@@ -1,19 +1,162 @@
 import React from 'react';
 import './App.css';
 
+import { Todo } from './types/todo';
+import { User } from './types/user';
+
 import users from './api/users';
+import todos from './api/todos';
+
+import TodoList from './components/TodoList';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+type State = {
+  todos: Todo[];
+  users: User[];
+  title: string;
+  selectedUser: string;
+  userValidation: boolean;
+  titleValidation: boolean;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todos,
+    users,
+    title: '',
+    selectedUser: '',
+    userValidation: true,
+    titleValidation: true,
+  };
+
+  changeHandle = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+    const { name, value } = e.target;
+
+    if (name === 'title') {
+      this.setState({ titleValidation: true });
+    }
+
+    if (name === 'selectedUser') {
+      this.setState({ userValidation: true });
+    }
+
+    return this.setState(state => ({
+      ...state,
+      [name]: value,
+    }));
+  };
+
+  addTask = () => {
+    const { title, selectedUser } = this.state;
+    const userFromSelect = this.state.users.find(user => user.name === selectedUser);
+
+    if (!userFromSelect || !title.trim().length) {
+      return this.Error();
+    }
+
+    const newTodo: Todo = {
+      userId: userFromSelect.id,
+      id: this.state.todos.length + 1,
+      completed: false,
+      title,
+    };
+
+    return this.setState((state) => ({
+      todos: [...state.todos, newTodo],
+      title: '',
+      selectedUser: '',
+    }));
+  };
+
+  Error = () => {
+    if (!this.state.title.trim().length) {
+      this.setState({
+        titleValidation: false,
+      });
+    }
+
+    if (!this.state.selectedUser.length) {
+      this.setState({
+        userValidation: false,
+      });
+    }
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <form
+          className="form"
+          action=""
+          method="GET"
+          onSubmit={e => (e.preventDefault())}
```

Comment:
move e.preventDefault inside addTask

```suggestion
          onSubmit={this.addTask}
```